### PR TITLE
Allow the reason for locking to be specified in m_lockserv.

### PR DESCRIPTION
**Pull request by SaberUK**
Reference: #1177

### CODE REVIEW

File: src/modules/m_lockserv.cpp [hunk 1]
```diff
@@ -86,31 +86,26 @@ class ModuleLockserv : public Module
...
+			ServerInstance->Users->QuitUser(user, locked);
 			return MOD_RES_DENY;
 		}
 		return MOD_RES_PASSTHRU;
 	}
 
 	ModResult OnCheckReady(LocalUser* user) CXX11_OVERRIDE
 	{
-		return locked ? MOD_RES_DENY : MOD_RES_PASSTHRU;
+		return locked.size() ? MOD_RES_DENY : MOD_RES_PASSTHRU;
```
Comment (attilamolnar) [inline src/modules/m_lockserv.cpp]: `!locked.empty()`

File: src/modules/m_lockserv.cpp [hunk 2]
```diff
@@ -52,23 +52,23 @@ class CommandLockserv : public Command
...
 
  public:
-	CommandUnlockserv(Module* Creator, bool &lock) : Command(Creator, "UNLOCKSERV", 0), locked(lock)
+	CommandUnlockserv(Module* Creator, std::string& lock) : Command(Creator, "UNLOCKSERV", 0, 1), locked(lock)
 	{
 		flags_needed = 'o';
 	}
 
-	CmdResult Handle (const std::vector<std::string> &parameters, User *user)
+	CmdResult Handle(const std::vector<std::string>& parameters, User* user)
```
Comment (attilamolnar) [inline src/modules/m_lockserv.cpp]: Please fix the code style in a separate commit

Comment (attilamolnar) [inline src/modules/m_lockserv.cpp]: (if you fix at all)

File: src/modules/m_lockserv.cpp [hunk 3]
```diff
@@ -27,23 +27,23 @@
...
 	{
-		if (locked)
+		if (!locked.empty())
 		{
 			user->WriteNotice("The server is already locked.");
 			return CMD_FAILURE;
 		}
 
-		locked = true;
+		locked = parameters.empty() || parameters[0].empty() ? "Server is temporarily closed. Please try again later." : parameters[0];
```
Comment (attilamolnar) [inline src/modules/m_lockserv.cpp]: Condition should be in ()

Comment (SaberUK) [inline src/modules/m_lockserv.cpp]: I switched to using `allow_empty_last_param = false` so the second check is unnecessary

**Comment by attilamolnar** [comment]
Thanks!

**Pull request merged**